libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -49,10 +189,33 @@ void CodeTransform::operator()(VariableDeclaration const& _varDecl)
 		while (variablesLeft--)
 			m_assembly.appendConstant(u256(0));
 	}
-	for (auto const& variable: _varDecl.variables)
+	int varIndex = numVariables - 1;
+	if (m_allowStackOpt)
+		// Try to find slots to re-use (top-to-bottom)
+		for (; varIndex >= 0; --varIndex)
+		{
+			auto& var = boost::get<Scope::Variable>(m_scope->identifiers.at(_varDecl.variables[varIndex].name));
+			if (!unreferenced(var))
+			{
+				if (m_unusedStackSlots.empty())
+					break;
+				int slot = *m_unusedStackSlots.begin();
+				m_unusedStackSlots.erase(m_unusedStackSlots.begin());
+				m_context->variableStackHeights[&var] = slot;
+				m_assembly.setSourceLocation(_varDecl.location);
+				if (int heightDiff = variableHeightDiff(var, true))
```

Comment:
Can we be sure that the variable is accessible here? Otherwise ``variableHeightDiff`` might throw - I could imagine that that could happen if there are a lot of variables, but I may be wrong.

Reply:
What do you mean by accessible? If it is too deep inside the stack, then `swapInstruction` below will throw and there is nothing we can do about it (for now). All other errors should be assertion errors.

Reply:
Yes, that's what I meant - and ``variableHeightDiff`` will already throw in that case - but shouldn't (and couldn't) we avoid that by just *not* trying to reuse that slot in that case?

Reply:
Ah, now I understand what you mean! Can you come up with a test where this would happen?

Reply:
Let me have a look, I'll try.

Reply:
So far I couldn't trigger this - it may be that since a variable is only freed at its last access (for which it can't be too deep in the stack anyways) and the very next variable will reuse its slot (which at that point still won't be too deep) that this can't happen, since in all these cases where it could happen we would already have accessed something too deep on the stack earlier anyways (and we can't do anything but throw there already) - but I'll recheck.

Reply:
So can we merge it like that for now?